src/util/TimeUtil.js

```diff
@@ -101,32 +101,39 @@ export default {
    * Conversion from a Date object to a locale string that mimics Okta's `short-with-timezone` format
    * but rounded down to the nearest hour
    * e.g. new Date(2024-09-05T00:00:00.000Z) -> 09/05/2024, 8:00 PM EDT
-   *
+   * 
    * @param {Date} date The Date object for the grace period expiry
-   * @param {LanguageCode} languageCode The user's language code / locale
+   * @param {string[] | string | null | undefined} languageTags The user's language tags, ordered by preference
    * @param {RoundDownToNearestHour} boolean Whether to round down to nearest hour. Defaults to true
    * @return {string} The formatted `short-with-timezone` local string
    */
-  formatDateToDeviceAssuranceGracePeriodExpiryLocaleString: (date, languageCode, roundDownToNearestHour = true) => {
+  formatDateToDeviceAssuranceGracePeriodExpiryLocaleString: (date, languageTags, roundDownToNearestHour = true) => {
     try {
-    // Invalid Date objects will return NaN for valueOf()
-      if (date && !isNaN(date.valueOf()) && languageCode !== null) {
-        // eslint-disable-next-line max-depth
-        if (roundDownToNearestHour) {
-          // Round down the date to the nearest hour
-          date.setMinutes(0, 0, 0);
-        }
-        return date.toLocaleString(languageCode, {
-          year: 'numeric',
-          month: '2-digit',
-          day: '2-digit',
-          hour: '2-digit',
-          minute: '2-digit',
-          timeZoneName: 'short',
-        });
-      } else {
+      // Invalid Date objects will return NaN for valueOf()
+      if (!(date instanceof Date) || isNaN(date.valueOf())) {
+        return null;
+      }
+      if (languageTags === null) {
         return null;
       }
+      if (roundDownToNearestHour) {
+        // Round down the date to the nearest hour
+        date.setMinutes(0, 0, 0);
+      }
+      const formatOptions = {
+        year: 'numeric',
+        month: '2-digit',
+        day: '2-digit',
+        hour: '2-digit',
+        minute: '2-digit',
+        timeZoneName: 'short',
+      };
+      if (languageTags === undefined) {
+        return date.toLocaleString(undefined, formatOptions);
+      }
+      const languageTagsPrefs = Array.isArray(languageTags) ? languageTags : [languageTags];
+
+      return date.toLocaleString(languageTagsPrefs, formatOptions);
```

Comment:
Might be able to consolidate this since `Date.prototype.toLocaleString` can accept `string | string[]` as the first parameter, so if the normalization of `languageTags` to an array isn't needed then this can be simplified to:
```suggestion
      return date.toLocaleString(languageTags, formatOptions);
```